core/src/cluster_info.rs

```diff
@@ -2025,31 +2032,38 @@ impl ClusterInfo {
     }
 
     fn process_packets(
-        &self,
+        self: Arc<Self>,
         requests: Vec<Packets>,
         thread_pool: &ThreadPool,
-        recycler: &PacketsRecycler,
-        response_sender: &PacketSender,
-        stakes: HashMap<Pubkey, u64>,
+        recycler: Arc<PacketsRecycler>,
+        response_sender: Arc<Mutex<PacketSender>>,
+        stakes: Arc<RwLock<HashMap<Pubkey, u64>>>,
         epoch_time_ms: u64,
     ) {
-        let sender = response_sender.clone();
-        thread_pool.install(|| {
-            requests.into_par_iter().for_each_with(sender, |s, reqs| {
-                self.handle_packets(&recycler, &stakes, reqs, s, epoch_time_ms)
+        for reqs in requests {
+            let new_self = self.clone();
+            let recycler2 = recycler.clone();
+            let stakes2 = stakes.clone();
+            let response_sender = response_sender.clone();
+            thread_pool.spawn(move || {
+                new_self.handle_packets(&recycler2, &stakes2, reqs, response_sender, epoch_time_ms)
```

Comment:
do we want to break up the `reqs` into individual requests to make it more fine grained?

Reply:
Each spawn has overhead, so you want to do as few as possible while still having decent execution overlap. One better strategy could be to look at the total number, if it's more than 4-8 * num_threads, then split them up into total / num_threads size batches. You would end up combining smaller ones and then splitting larger ones. Just didn't get to it yet.